dataclients/kubernetes/kube.go

```diff
@@ -417,26 +438,95 @@ func (c *Client) ingressToRoutes(items []*ingressItem) []*eskip.Route {
 			// currently handled as mandatory
 			host := []string{"^" + strings.Replace(rule.Host, ".", "[.]", -1) + "$"}
 
+			// update Traffic field for each backend
+			computeBackendWeights(backendWeights, rule)
+
 			for _, prule := range rule.Http.Paths {
-				r, err := c.convertPathRule(i.Metadata.Namespace, i.Metadata.Name, rule.Host, prule, servicesURLs)
-				if err != nil {
-					// tolerate single rule errors
-					//
-					// TODO:
-					// - check how to set failures in ingress status
-					log.Errorf("error while getting service: %v", err)
-					continue
+				if prule.Backend.Traffic > 0 {
+					r, err := c.convertPathRule(i.Metadata.Namespace, i.Metadata.Name, rule.Host, prule, servicesURLs)
+					if err != nil {
+						// tolerate single rule errors
+						//
+						// TODO:
+						// - check how to set failures in ingress status
+						log.Errorf("error while getting service: %v", err)
+						continue
+					}
+
+					r.HostRegexps = host
+					routes = append(routes, r)
 				}
-
-				r.HostRegexps = host
-				routes = append(routes, r)
 			}
 		}
 	}
 
 	return routes
 }
 
+// computeBackendWeights computes and sets the backend traffic weights on the
+// rule backends.
+// The traffic is calculated based on the following rules:
+//
+// * if no weight is defined for a backend it will get weight 0.
+// * if no weights are specified for all backends of a path, then traffic will
+//   be distributed equally.
+//
+// Each traffic weight is relative to the number of backends per path. If there
+// are multiple backends per path the weight will be relative to the number of
+// remaining backends for the path e.g. if the weight is specified as
+//
+//      backend-1: 0.2
+//      backend-2: 0.6
+//      backend-3: 0.2
+//
+// then the weight will be calculated to:
+//
+//      backend-1: 0.2
+//      backend-2: 0.75
+//      backend-3: 1.0
+//
+// where for a weight of 1.0 no Traffic predicate will be generated.
+func computeBackendWeights(backendWeights map[string]float64, rule *rule) {
```

Comment:
> if I understand correctly, the calculated weights assume that the leave routes on the same path are evaluated in the order of the input. Is this correct?

Yeah, this was my assumption, would changing to a stable sort cause problems elsewhere?